### PR TITLE
Update 000-waterfox.js

### PR DESCRIPTION
**Pull request by Peacock365**
- Removed duplicate browser.newtabpage.activity-stream.feeds.section.topstories
- Emptied extensions.getAddons.langpacks.url URL string (reason: Waterfox bundles its own language packs).
- Emptied extensions.recommendations.privacyPolicyUrl URL string (reason: the feature is disabled in Waterfox)
- Set app.update.BITS.enabled to "false" (reason: Waterfox doesn't use the BITS updating mechanism)
- Set browser.urlbar.speculativeConnect.enabled to "false" (reason: get rid of unsolicited connections, align with Waterfox G3)
- Set browser.urlbar.sponsoredTopSites to "false" (reason: get rid of Mozilla advertisements)
- Set browser.xul.error_pages.expert_bad_cert to "true" (reason: Give users more flexibility in case of bad certificate)
- Set plugin.default.state to "1" (reason: Click-to-play is desirable for plugins as a middle ground between security and usability)
- Set plugin.state.flash to "1" (reason: same as for plugin.default.state above)
- Emptied breakpad.reportURL URL string (reason: get rid of crash reports going to Mozilla)
- Emptied toolkit.crashreporter.infoURL URL string (reason: crash reporter is inactive in Waterfox)
- Set services.sync.prefs.sync.app.shield.optoutstudies.enabled to "false" (reason: Waterfox doesn't use Firefox Experiments)
- Set services.sync.prefs.sync.browser.crashReports.unsubmittedCheck.autoSubmit2 to "false" (reason: crash reporter is inactive in Waterfox)
- Set services.sync.prefs.sync.browser.discovery.enabled to "false" (reason: Discovery is inactive in Waterfox)
- Set services.sync.prefs.sync.browser.newtabpage.activity-stream.asrouter.userprefs.cfr.addons to "false" (reason: No add-on recommendations in Waterfox)
- Set services.sync.prefs.sync.browser.newtabpage.activity-stream.asrouter.userprefs.cfr.features to "false" (reason: No feature recommendations in Waterfox)
- Set services.sync.prefs.sync.browser.newtabpage.activity-stream.feeds.section.topstories to "false" (reason: Top Stories are disabled in Waterfox)
- Set services.sync.prefs.sync.browser.newtabpage.activity-stream.section.topstories.rows to "false" (reason: Since you don't have Top Stories, you also don't need their rows)
- Set services.sync.prefs.sync.browser.newtabpage.activity-stream.section.highlights.includePocket to "false" (reason: no Pocket in Waterfox)
- Emptied browser.newtabpage.activity-stream.discoverystream.locale-list-config (reason: Waterfox has Discovery Stream disabled)
- Emptied browser.newtabpage.activity-stream.discoverystream.region-stories-config (reason: Waterfox has Discovery Stream disabled)
- Emptied browser.newtabpage.activity-stream.discoverystream.region-spocs-config (reason: Waterfox has Discovery Stream disabled)
- Set browser.newtabpage.activity-stream.discoverystream.spocs.personalized to "false" (reason: Waterfox has Discovery Stream disabled)
- Set browser.messaging-system.whatsNewPanel.enabled to "false" (reason: Get rid of the annoying What's New Panel that only concerns Mozilla releases)
- Emptied messaging-system.rsexperimentloader.collection_id (reason: No FF Experiments in Waterfox)
- Set security.secure_connection_icon_color_gray to "false" (reason: green lock icon ensures better visibility of the fact that the connection is encrypted)
- Set security.insecure_connection_text.enabled to "true" (reason: ensure better visibility of the fact that a connection is not encrypted)
- Set security.insecure_connection_text.pbmode.enabled to "true" (reason: ensure better visibility of the fact that a connection is not encrypted)
- Emptied identity.mobilepromo.android URL string (reason: no need to promote the Android version of Firefox)
- Emptied identity.mobilepromo.ios URL string (reason: no need to promote the iOS version of Firefox)
- Emptied toolkit.telemetry.server_owner (reason: no telemetry collected in Waterfox)
- Set privacy.userContext.enabled to "true" (reason: enable containers, align with Waterfox G3)
- Set privacy.userContext.ui.enabled to "true" (reason: enable containers, align with Waterfox G3)
- Set browser.tabs.crashReporting.sendReport to "false" (reason: crash reporter not desired in Waterfox)
- Set browser.tabs.crashReporting.includeURL to "false" (reason: crash reporter not desired in Waterfox)
- Set extensions.experiments.enabled to "true" (reason: enable support for experimental WebExtension APIs by default in Waterfox)
- Set extensions.pocket.onSaveRecs to "false" (reason: Pocket is not desirable in Waterfox)
- Emptied extensions.pocket.onSaveRecs.locales (reason: Pocket is not desirable in Waterfox)
- Set browser.crashReports.unsubmittedCheck.enabled to "false" (reason: crash reporter not desired in Waterfox)
- Set browser.crashReports.unsubmittedCheck.chancesUntilSuppress to "0" (reason: crash reporter not desired in Waterfox)
- Set browser.crashReports.unsubmittedCheck.autoSubmit2 to "false" (reason: crash reporter not desired in Waterfox)
- Emptied app.normandy.api_url URL string (reason: Firefox Experiments not desired in Waterfox)
- Set app.normandy.dev_mode to "false" (reason: Firefox Experiments not desired in Waterfox)
- Set app.normandy.enabled to "false" (reason: Firefox Experiments not desired in Waterfox)
- Set app.normandy.first_run to "false" (reason: Firefox Experiments not desired in Waterfox)
- Set app.normandy.logging.level to "50" (reason: Firefox Experiments not desired in Waterfox)
- Set app.normandy.run_interval_seconds to "-1" (reason: Firefox Experiments not desired in Waterfox)
- Emptied app.normandy.shieldLearnMoreUrl URL string (reason: Firefox Experiments not desired in Waterfox)
- Emptied app.normandy.last_seen_buildid (reason: Firefox Experiments not desired in Waterfox)
- Set app.normandy.onsync_skew_sec to "-1" (reason: Firefox Experiments not desired in Waterfox)
- Set app.shield.optoutstudies.enabled to "false" (reason: Firefox Experiments not desired in Waterfox)
- Set toolkit.coverage.enabled to "false" (reason: get rid of the coverage telemetry ping)
- Emptied toolkit.coverage.endpoint.base URL string (reason: get rid of the coverage telemetry ping)
- Set toolkit.coverage.opt-out to "true" (reason: get rid of the coverage telemetry ping)
- Set browser.discovery.enabled to "false" (reason: Extension discovery panel is disabled anyway)
- Set browser.discovery.containers.enabled to "false" (reason: Extension discovery panel is disabled anyway)
- Emptied browser.discovery.sites URL string (reason: Extension discovery panel is disabled anyway)
- Set default-browser-agent.enabled to "false" (reason: get rid of the scheduled telemetry task)

### CODE REVIEW

**Comment by Peacock365** [comment]
@MrAlex94 @adamp01 

Please review.

**Comment by MrAlex94** [comment]
Already on the todo list. Unfortunately a lot of the preferences are redundant. Setting or changing them makes no difference since the code to ingest those values will never execute. I will comb through and only take the necessary ones, if there are any.

**Comment by Peacock365** [comment]
> Already on the todo list. Unfortunately a lot of the preferences are redundant. Setting or changing them makes no difference since the code to ingest those values will never execute. I will comb through and only take the necessary ones, if there are any.

I did some DiD, sometimes for optics' sake. I know that some are redundant, but avoid confusion for users. And the list doesn't necessarily have to be as short as possible, I think.

File: browser/app/profile/000-waterfox.js [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+pref("browser.safebrowsing.provider.google4.updateURL", "", locked);
+pref("browser.safebrowsing.provider.mozilla.gethashURL", "", locked);
+pref("browser.safebrowsing.provider.mozilla.updateURL", "", locked);
+pref("browser.send_pings", false, locked);
+pref("browser.tabs.remote.separatedMozillaDomains", "", locked);
+pref("browser.topsites.contile.enabled", false, locked);
+pref("browser.topsites.contile.endpoint", "", locked);
+pref("browser.topsites.useRemoteSetting", false, locked);
+pref("browser.uiCustomization.state", "{\"placements\":{\"widget-overflow-fixed-list\":[],\"nav-bar\":[\"back-button\",\"forward-button\",\"stop-reload-button\",\"urlbar-container\",\"save-to-pocket-button\",\"downloads-button\",\"library-button\",\"fxa-toolbar-menu-button\"],\"TabsToolbar\":[\"tabbrowser-tabs\",\"new-tab-button\",\"alltabs-button\"],\"PersonalToolbar\":[\"import-button\",\"personal-bookmarks\"]},\"seen\":[\"profiler-button\",\"developer-button\"],\"dirtyAreaCache\":[\"nav-bar\",\"PersonalToolbar\"],\"currentVersion\":17,\"newElementCount\":3}");
+pref("browser.uidensity", 1, locked);
```
Comment (steanne) [inline browser/app/profile/000-waterfox.js:105]: please don't lock this.  setting a default is one thing, but forcing everyone to a density that may not suit their system is not reasonable.

**Comment by Peacock365** [comment]
@MrAlex94 

Please consider merging my commit, unless there is any disagreement with any of the changes I made. Happy to discuss potential issues if you see any. 

Actually, I thought you would merge this ahead of the Waterfox G4.0.0 release, because as it stands, Waterfox still has Firefox Experiments enabled which I do not find acceptable and which my commit (among other things) would fix. I expected you to release Waterfox G3.2.7 first and anticipated Waterfox G4.0.0 for early November to be honest. Currently, Waterfox G4.0.0 also still contains DoH Rollout as a system add-on which is hijacking the DNS provider without user approval, that too is unacceptable IMHO. The release feels a bit rushed and I don't really know why, Firefox 78 ESR support was only going to end in early November...

**Comment by MrAlex94** [comment]
We've had quite a few issues caused by preference configurations alone - so we aren't very keen on just blindly accepting these all without checking what they do.

DoH has been disabled (it's the extention that enables it I believe?) due to causing issues elsewhere.

Flipping the `*.sync.*` is pointless to be honest as prefs syncing is off by default - and also experiments should be disabled elsewhere making these pref flips just for appearances.

Some others you have added are good to flip though - so will get to them in due time.

**Comment by MrAlex94** [comment]
Have gone through the rest of your pref changes and added what I think to be the most appropriate in https://github.com/WaterfoxCo/Waterfox/commit/f8daa94aead9e8a0b6d91c038b736b399669d46d.

**Pull request closed without merging**